From 8735125c2fffb6555068a3e50243b8ed07f94a4e Mon Sep 17 00:00:00 2001
From: Aloka Dixit <quic_alokad@quicinc.com>
Date: Mon, 11 Dec 2023 11:19:08 -0800
Subject: [PATCH 2/2] wifi: ath12k: add stats for SW based progressive DFS

Add support to log following new statistics related to the device
bandwidth parameters:
	current_operating_width,
	current_device_width,
	last_radar_type,
	dfs_reg_domain,
	radar_mask_bit,
	radar_rssi,
	radar_dfs_flags,
	band_center_frequency_operating,
	band_center_frequency_device

Signed-off-by: Aloka Dixit <quic_alokad@quicinc.com>
Signed-off-by: Shivani Tambatkar <quic_stambatk@quicinc.com>
---
 .../wireless/ath/ath12k/debugfs_htt_stats.c   | 20 +++++++++++++++++++
 .../wireless/ath/ath12k/debugfs_htt_stats.h   |  9 +++++++++
 2 files changed, 29 insertions(+)

--- a/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
+++ b/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
@@ -7465,6 +7465,26 @@ static inline void htt_print_phy_stats_t
 			 htt_stats_buf->radar_cs_cnt);
 	len += scnprintf(buf + len, buf_len - len, "ani_level = %d\n",
 			 htt_stats_buf->ani_level);
+
+	len += scnprintf(buf + len, buf_len - len, "current operating bw = %u\n",
+			 htt_stats_buf->current_operating_width);
+	len += scnprintf(buf + len, buf_len - len, "current device bw = %u\n",
+			 htt_stats_buf->current_device_width);
+	len += scnprintf(buf + len, buf_len - len, "last radar type = %u\n",
+			 htt_stats_buf->last_radar_type);
+	len += scnprintf(buf + len, buf_len - len, "dfs regulatory domain = %u\n",
+			 htt_stats_buf->dfs_reg_domain);
+	len += scnprintf(buf + len, buf_len - len, "radar mask bit = %u\n",
+			 htt_stats_buf->radar_mask_bit);
+	len += scnprintf(buf + len, buf_len - len, "radar rssi = %d\n",
+			 htt_stats_buf->radar_rssi);
+	len += scnprintf(buf + len, buf_len - len, "radar dfs flags = %u\n",
+			 htt_stats_buf->radar_dfs_flags);
+	len += scnprintf(buf + len, buf_len - len, "operating center freq = %u\n",
+			 htt_stats_buf->band_center_frequency_operating);
+	len += scnprintf(buf + len, buf_len - len, "device center freq = %u\n",
+			 htt_stats_buf->band_center_frequency_device);
+
 	len += scnprintf(buf + len, buf_len - len,
 			 "=================================================\n");
 
--- a/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.h
+++ b/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.h
@@ -3654,6 +3654,15 @@ struct htt_phy_stats_tlv {
 	s32 ani_level;
 	u32 fw_run_time;
 	s32 runtime_nf_chain[HTT_STATS_MAX_CHAINS];
+	u32 current_operating_width;
+	u32 current_device_width;
+	u32 last_radar_type;
+	u32 dfs_reg_domain;
+	u32 radar_mask_bit;
+	s32 radar_rssi;
+	u32 radar_dfs_flags;
+	u32 band_center_frequency_operating;
+	u32 band_center_frequency_device;
 };
 
 struct htt_phy_reset_counters_tlv {
